AEPMessaging/Sources/Messaging+PublicAPI.swift

```diff
@@ -49,40 +48,41 @@ import UserNotifications
                           source: EventSource.requestContent,
                           data: eventData)
         MobileCore.dispatch(event: event)
-        return true
     }
 
     /// Sends the push notification interactions as an experience event to Adobe Experience Edge.
     /// - Parameter response: UNNotificationResponse object which contains the payload and xdm informations.
-    /// - Returns: boolean value that signifies whether the notification originated from AJO and whether the API has proceeded with notification tracking.
-    @discardableResult
-    static func handleNotificationResponse(_ response: UNNotificationResponse) -> Bool {
+    static func handleNotificationResponse(_ response: UNNotificationResponse, closure: ((MessagingPushTrackingStatus) -> Void)? = nil) {
         let notificationRequest = response.notification.request
 
         // Checking if the message has the _xdm key that contains tracking information
         guard let xdm = notificationRequest.content.userInfo[MessagingConstants.XDM.AdobeKeys._XDM] as? [String: Any], !xdm.isEmpty else {
             Log.debug(label: MessagingConstants.LOG_TAG, "XDM specific fields are missing from push notification response. Ignoring to track push notification.")
-            return false
+            closure?(.noTrackingData)
+            return
         }
 
-        DispatchQueue.global().async {
-            hasApplicationOpenedForResponse(response, completion: { isAppOpened in
+        hasApplicationOpenedForResponse(response, completion: { isAppOpened in
```

Comment:
The bg thread was introduced when this function had to return boolean in this PR 
https://github.com/adobe/aepsdk-messaging-ios/pull/207/files#diff-c180dc7b3b85d0fca4667cd33284c2d8efd80f5c481608665740a3e5882cd78dR68. Because we call an async function `hasApplicationOpenedForResponse`.

With the return value callback, we dont to switch to background thread to dispatch an event.

Reply:
i think it's a good idea to get off the calling thread as soon as possible here, considering the app developer may be calling it from the UI thread

Reply:
Yeah that makes sense, given that we call system API and not just marshall and directly dispatch an event